openapi-validator/src/main/java/io/ballerina/openapi/validator/ParameterValidator.java

```diff
@@ -0,0 +1,200 @@
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static io.ballerina.openapi.validator.Constants.ARRAY;
+import static io.ballerina.openapi.validator.Constants.SQUARE_BRACKETS;
+import static io.ballerina.openapi.validator.TypeValidatorUtils.convertBallerinaType;
+import static io.ballerina.openapi.validator.TypeValidatorUtils.convertOpenAPITypeToBallerina;
+import static io.ballerina.openapi.validator.ValidatorUtils.extractReferenceType;
+import static io.ballerina.openapi.validator.ValidatorUtils.getNormalizedPath;
+import static io.ballerina.openapi.validator.ValidatorUtils.getNumberFormatType;
+import static io.ballerina.openapi.validator.ValidatorUtils.reportDiagnostic;
+import static io.ballerina.openapi.validator.ValidatorUtils.unescapeIdentifier;
+
+/**
+ * This contains the all parameter validation.
+ *
+ * @since 1.1.0
+ */
+public class ParameterValidator extends AbstractMetaData implements SectionValidator, Validator {
+    private final Map<String, Node> parameters;
+    private final List<Parameter> oasParameters;
+    // This default location is map to relevant resource function
+
+
+    public ParameterValidator(MetaData metaData, Map<String, Node> parameters, List<Parameter> oasParameters) {
+        super(metaData.getContext(), metaData.getOpenAPI(), metaData.getPath(), metaData.getMethod(),
+                metaData.getSeverity(), metaData.getLocation());
+        this.parameters = parameters;
+        this.oasParameters = oasParameters;
+    }
+
+    @Override
+    public void validate() {
+        //Ballerina to OAS parameter validation , here we validate query, path parameters.
+        validateBallerina();
+
+        //OAS->Ballerina parameter validate
+        if (oasParameters != null) {
+            validateOpenAPI();
+        }
+    }
+
+    /**
+     * Validate OAS parameter against ballerina parameters.
+     */
+    @Override
+    public void validateOpenAPI() {
+
+        oasParameters.forEach(parameter -> {
+            if (parameter.get$ref() != null) {
+                Optional<String> parameterName = extractReferenceType(parameter.get$ref());
+                if (parameterName.isEmpty()) {
+                    return;
+                }
+                parameter = openAPI.getComponents().getParameters().get(parameterName.get());
+            }
+            if (!(parameter instanceof HeaderParameter) ||
+                    parameter.getIn() != null && !parameter.getIn().equals("header")) {
+                // headerValidation
+                AtomicBoolean isImplemented = new AtomicBoolean(false);
+                Parameter finalParameter1 = parameter;
+                parameters.forEach((paramName, paramNode) -> {
+                    // avoid headers
+                    if (finalParameter1.getName().equals(paramName)) {
+                        isImplemented.set(true);
+                    }
+                });
```

Comment:
shall we convert this to a for loop so that we wont need an additional final parameters + atomic variables?